### PR TITLE
feat(metric): track aredis.ConnectionError

### PR DESCRIPTION
**Pull request by sileht**
Fixes MRGFY-363

### CODE REVIEW

File: mergify_engine/web/root.py [hunk 1]
```diff
@@ -66,6 +67,7 @@ async def shutdown() -> None:
 async def redis_errors(
     request: requests.Request, exc: aredis.exceptions.ConnectionError
 ) -> responses.JSONResponse:
+    statsd.increment("engine.redis.connection-errors")
```
Comment (jd) [inline mergify_engine/web/root.py:70]:
```suggestion
    statsd.increment("redis.client.connection.errors")
```
Then we can use this in every service and apply tags.

File: mergify_engine/worker.py [hunk 2]
```diff
@@ -242,6 +242,9 @@ async def _translate_exception_to_retries(
         try:
             yield
         except Exception as e:
+            if isinstance(e, aredis.exceptions.ConnectionError):
+                statsd.increment("engine.redis.connection-errors")
```
Comment (jd) [inline mergify_engine/worker.py:246]: There's a lot of repetition, we might forget to add this somewhere. Can we wrap the `aredis` client to take care of that?

**Review by jd — DISMISSED** [review]

File: mergify_engine/worker.py (hunk 2 above)
Comment (jd) [reply mergify_engine/worker.py]: @sileht should I wait for this or you're giving up?

Comment (sileht) [reply mergify_engine/worker.py]: Oh, I miss it, but I don't see how to do that properly without involving some python black magic. ConnectionError can be raised everywhere in aredis code.

**Review by jd — APPROVED** [review]

**Pull request merged**